app/services/plans/update_service.rb

```diff
@@ -2,70 +2,44 @@
-      #       is attached to subscriptions
-      unless plan.attached_to_subscriptions?
-        plan.code = args[:code]
-        plan.interval = args[:interval].to_sym
-        plan.pay_in_advance = args[:pay_in_advance]
-        plan.amount_cents = args[:amount_cents]
-        plan.amount_currency = args[:amount_currency]
-        plan.trial_period = args[:trial_period]
-        plan.bill_charges_monthly = args[:interval].to_sym == :yearly ? args[:bill_charges_monthly] || false : nil
-      end
-
-      metric_ids = args[:charges].map { |c| c[:billable_metric_id] }.uniq
-      if metric_ids.present? && plan.organization.billable_metrics.where(id: metric_ids).count != metric_ids.count
-        return result.not_found_failure!(resource: 'billable_metrics')
-      end
-
-      ActiveRecord::Base.transaction do
-        plan.save!
-
-        process_charges(plan, args[:charges])
-      end
+    def self.call(...)
+      new(...).call
+    end
 
-      result.plan = plan
-      result
-    rescue ActiveRecord::RecordInvalid => e
-      result.record_validation_failure!(record: e.record)
+    def initialize(plan:)
+      @plan = plan
+      super
     end
 
-    def update_from_api(organization:, code:, params:)
-      plan = organization.plans.find_by(code: code)
+    def call(args)
       return result.not_found_failure!(resource: 'plan') unless plan
 
-      plan.name = params[:name] if params.key?(:name)
-      plan.description = params[:description] if params.key?(:description)
+      plan.name = args[:name] if args.key?(:name)
+      plan.description = args[:description] if args.key?(:description)
 
+      # NOTE: Only name and description are editable if plan
+      #       is attached to subscriptions
       unless plan.attached_to_subscriptions?
-        plan.code = params[:code] if params.key?(:code)
-        plan.interval = params[:interval].to_sym if params.key?(:interval)
-        plan.pay_in_advance = params[:pay_in_advance] if params.key?(:pay_in_advance)
-        plan.amount_cents = params[:amount_cents] if params.key?(:amount_cents)
-        plan.amount_currency = params[:amount_currency] if params.key?(:amount_currency)
-        plan.trial_period = params[:trial_period] if params.key?(:trial_period)
-        plan.bill_charges_monthly = params[:interval]&.to_sym == :yearly ? params[:bill_charges_monthly] || false : nil
+        plan.code = args[:code] if args.key?(:code)
+        plan.interval = args[:interval].to_sym if args.key?(:interval)
+        plan.pay_in_advance = args[:pay_in_advance] if args.key?(:pay_in_advance)
+        plan.amount_cents = args[:amount_cents] if args.key?(:amount_cents)
+        plan.amount_currency = args[:amount_currency] if args.key?(:amount_currency)
+        plan.trial_period = args[:trial_period] if args.key?(:trial_period)
+        plan.bill_charges_monthly = args[:interval]&.to_sym == :yearly ? args[:bill_charges_monthly] || false : nil
```

Comment:
Style/TernaryParentheses: Use parentheses for ternary expressions with complex conditions.